src/mpi/simulation.rs

```diff
@@ -80,6 +84,18 @@ impl Simulation {
                     agent.curr_act().act_type.clone(),
                 ),
             );
+
+            if self.router.is_some() {
+                let end_activity = agent.curr_act();
+                let new_activity = agent.next_act();
```

Comment:
This is really picky now, but `curr_act` and `next_act` would have been my first guess as variable names. Why invent new names...